dockers/docker-dhcp-server/dhcpservd.py

```diff
@@ -0,0 +1,521 @@
+        }
+        self.ip_ports = {
+            IPv4Network('192.168.0.0/24')
+        }
+        Args:
+            port_ipv4: Table object.
+            vlan_interfaces: Vlan information, sample:
+                {
+                    'Vlan1000': [{
+                        'network': IPv4Network('192.168.0.0/24'),
+                        'ip': '192.168.0.1/24'
+                    }]
+                }
+        """
+        self.port_ips = {}
+        self.ip_ports = {}
+        for port_key in port_ipv4.getKeys():
+            port_config = self.get_entry(port_ipv4, port_key)
+            # Cannot specify both 'ips' and 'ranges'
+            if "ips" in port_config and len(port_config["ips"]) != 0 and "ranges" in port_config \
+               and len(port_config["ranges"]) != 0:
+                syslog.syslog(syslog.LOG_WARNING, f"Port config for {port_key} contains both ips and ranges, skip")
+                continue
+            splits = port_key.split("|")
+            # Get dhcp interface name like Vlan1000
+            dhcp_interface_name = splits[0]
+            # Get dhcp member interface name like etp1
+            port = self.port_alias_map[splits[1]]
+            if dhcp_interface_name not in self.port_ips:
+                self.port_ips[dhcp_interface_name] = {}
+            # Get ip information of Vlan
+            dhcp_interface = vlan_interfaces[dhcp_interface_name]
+
+            for dhcp_interface_ip in dhcp_interface:
+                self.ip_ports[dhcp_interface_ip["network"]] = dhcp_interface_name
+
+            if "ips" in port_config and len(port_config["ips"]) != 0:
+                for ip in set(port_config["ips"].split(",")):
+                    ip_address = ipaddress.ip_address(ip)
+                    # Loop the IP of the dhcp interface and find the network that target ip is in this network.
+                    self.match_range_network(dhcp_interface, dhcp_interface_name, port, [ip_address, ip_address])
+            if "ranges" in port_config and len(port_config["ranges"]) != 0:
+                for range_name in set(port_config["ranges"].split(",")):
+                    if range_name not in self.ranges:
+                        syslog.syslog(syslog.LOG_WARNING, f"Range {range_name} is not in range table, skip")
+                        continue
+                    range = self.ranges[range_name]
+                    # Loop the IP of the dhcp interface and find the network that target range is in this network.
+                    self.match_range_network(dhcp_interface, dhcp_interface_name, port, range)
+        # Merge ranges to avoid overlap
+        for dhcp_interface_name, value in self.port_ips.items():
+            for dhcp_interface_ip, port_range in value.items():
+                for port_name, ip_range in port_range.items():
+                    self.port_ips[dhcp_interface_name][dhcp_interface_ip][port_name] = self.merge_intervals(ip_range)
+
+    def generate_kea_dhcp4_config(self):
```

Comment:
Can you extract cfggen related methods to a new class: `dhcp-cfggen` for pure config generation logic?

And here you can invoke the `dhcp-cfggen` to generate and dump to file system.